CONTRIBUTING.md

```diff
@@ -9,7 +7,68 @@ MongoDB welcomes community contributions! If you’re interested in making a con
+- After 30 more days of no activity we automatically close the issue / PR.
+
+## Prerequisites
+- [AWS CloudFormation CLI](https://github.com/aws-cloudformation/cloudformation-cli)
+- (Optional - only need if building from source) [AWS CloudFormation CLI Go Plugin](https://github.com/aws-cloudformation/cloudformation-cli-go-plugin/) > v1.0
+- (Optional - only need if building from source) [Go](https://golang.org/doc/install) v1.19
+
+## Testing the Provider
+Please see README for each resource for details on unit and integrated AWS testing.
+
+## Securing your Deployment
+This repository contains utilities which use both the AWS and MongoDB Atlas APIs to progamatically create, destory, or otherwise manage resources in the AWS and MongoDB clouds. This can have costs or other impact so please use these tools with diligence and follow best practices.
+
+The automated workflows require the following secrets to be placed into your Github forks' Settings/Secrets:
+
+```
+AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY
+AWS_REGION
+ATLAS_PUBLIC_KEY
+ATLAS_PRIVATE_KEY
+ATLAS_ORG_ID
+```
+
+## Security - Setup
+
+1. Create and note your [MongoDB Atlas API Key](https://www.mongodb.com/docs/atlas/configure-api-access/#std-label-atlas-admin-api-access).
+2. Create and note your AWS Access Key and AWS Secret Key ID.
+3. Follow the Github docs on how to [create a Secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository) in your clone of this repository.
+
+## Using the MongoDB Atlas CFN Resources
+
+The fastest way to use the resources is with the official [MongoDB Atlas on AWS Partner Solutions](https://aws.amazon.com/solutions/partners/mongodb-atlas/) CloudFormation templates. The [templates](https://github.com/aws-quickstart/quickstart-mongodb-atlas/templates) folder contains concrete CloudFormation templates you can use to start your own projects.
+
+There are two main parts of this project:
+
+1. [quickstart-mongodbatlas](quickstart-mongodbatlas) This is a mirror of https://github.com/aws-quickstart/quickstart-mongodb-atlas.
+
+2. [cfn-resources](cfn-resources) A set of AWS CloudFormation custom resource providers for MongoDB Atlas Resources. Currently, AWS requires users to manually deploy these resources in each AWS region one one desires to use them in. We support this workflow through the standard AWS `cfn submit` tooling. Scripts and Github actions are contained in this repository which demonstrate automating this deployment process.
+
+
+## Registering resources to run locally
+
+1. Please check that you satisfy all the [requirements](#Requirements) before proceeding.
+2. Clone this repo.
+3. A helper script `[cfn-resources/cfn-submit-helper.sh](cfn-submit-helper.sh)` will build and submit each resource for you. You can also run the `cfn submit` tool yourself. Note- this step will soon not be required when AWS launch a public registry.
```

Comment:
do users have to manually invoke the cfn-submit-helper script? It's not clear if this is something that happens automatically or requires user intervention. I think we might need to revise this step to make it more active (i.e. `Use the helper script to ...`)

Reply:
Agreed & updated